src/Controllers/organisation.controller.ts

```diff
@@ -219,13 +214,10 @@ export const deleteOrganisation = async (req: Request<DeleteOrganisationQueryPar
 
     res.status(204).end();
   } catch (e) {
-    if (e instanceof PrismaClientKnownRequestError) {
-      if ((e.code = "P2025")) {
-        return res.status(404).json(generateError(`The organisation with the ID ${pid} could not be found`));
-      }
-    } else if (e instanceof PrismaClientUnknownRequestError) {
-      return res.status(400).send(generateError("Unkonwn error occured. This could be due to malformed IDs"));
+    if (e instanceof PrismaClientKnownRequestError && e.code === "P2025") {
+      throw new NotFoundError("organisation", pid);
     }
+    throw e;
   }
 
   return res.status(500).json(genericError);
```

Comment:
Why does this exist? If I see this correctly, it will not be called.

Reply:
It should not be called no. It is actually called tho when everything _works_, eg. when line 220 is executed, because it does not return.